test/i18n: enforce regex translations at test time

quisquous opened the pull request:
This is working towards #1306.

Previously, the find missing translations script found these (although
it apparently missed some) and would not block merging in new errors.
This PR moves the regex translation to the test code, so `npm run test`
and github workflows will complain if they are missing.  As there are
now no more errors in this department after this PR, we can ratchet this
fixed so the followup can proceed.

The next steps here are to add code to popup-text to use replaceSync to
do the translation (if there are no locale regexes like netRegexDe).  For
backwards compat, the locale regexes will always be valid (and will
override this translation) in case any user triggers are using them in
weird ways.  However, once that it done we can deprecate locale regexes
and remove them from the documentation and triggers.

quisquous commented:
cc @Akurosia @valarnin because we've talked about #1306

quisquous commented on test/helper/test_trigger.ts:
```diff
@@ -26,6 +27,14 @@ import {
+
+// Set a global flag to mark regexes for NetRegexes.doesNetRegexNeedTranslation.
+// See details in that function for more information.
+NetRegexes.setFlagTranslationsNeeded(true);
```
Mmm, good thought!  I'll set it in `before` and unset it in `after`.

quisquous commented:
@Akurosia can you take a look at the translations? It seems like there's inconsistencies in Haukke Manor and t1, in terms of the replace section and the trigger regexes.  It seems like find missing translations didn't find these before because these files didn't have a timeline, and this check was in the timeline test.

Akurosia commented:
should these inconsistent translations be shown when running `npm run find-translations -- -l=de -f=ui` ? because atm i only get a few results from my latest PRs (that are not merged with this branch here)

Akurosia commented:
also i had a look at these, and do we need `.*?` in the end of a addedCombatant/ message ?

quisquous commented:
> should these inconsistent translations be shown when running npm run find-translations -- -l=de -f=ui ? because atm i only get a few results from my latest PRs (that are not merged with this branch here)

No, it's merged into test, so you'll get errors when you run `npm run test` or via the github workflow rather than needing an extra script to do it.

> also i had a look at these, and do we need .*? in the end of a addedCombatant message ?

You shouldn't.

Akurosia commented:
Ok maybe i misunderstood something, but i ran both find-translation and test and non of them showed any issues (see screenshot attached). I will go through all ARR and HW files to see if there are any `.*?` for e.g. addedCombatants or any other inconsistency.

![WindowsTerminal_9NK2ytYxFH](https://user-images.githubusercontent.com/7465469/167790175-51b6c67d-d0f9-4bdc-9fd1-4b019cbcfcd4.png)

quisquous commented:
Sorry, I feel like there's some confusion here.  With this patch, there should not be any issues, as I fixed them.  I was mostly asking for you to take a look at the translations I changed in this PR to make it pass `npm run test`.  If you take this PR but then remove all the changes to `ui/raidboss/data` then it will have a bunch of failures.  However, I'm not sure those translations were correct as the `timelineReplace` did not agree with the `netRegexFr` (etc).

Re: addedCombatant.  I do not know why some of the lines in t1 have a `.*?` on the added combatant.  Maybe this can/should be removed here.  In general, addedCombatant should not have/need this.  I think this can be changed after this.

quisquous commented:
WAIT, I didn't see you were targeting this PR with your PR.  Was that intentional?
